react-app/src/components/ActionIcon/index.js

```diff
@@ -24,7 +30,7 @@ export default class ActionIcon extends PureComponent {
     onClick: PropTypes.func.isRequired,
   };
 
-  _handleClick = event => this.props.onClick(event);
+  _handleClick = () => this.props.onClick(this.props.icon);
```

Comment:
Why would you return the icon?

Reply:
I think that passing the `icon` prop could be useful to avoid having two functions to handle header icon click events. I mean, instead of having something like:

```
onBack = () => {}
onApply = () => {}
```

We could have:

```
onHeaderEvent = icon => {
  if (icon === ActionIcon.BACK) {}
  else {}
}
```

Reply:
I have to mention that there isn't a real benefit of doing this having two buttons only. It's just a way of organize the containers in a different way.

Return the event has no sense. On the other hand, if the `icon` props is returned, both ways of implementing the event handlers in the containers are supported.